dotCMS/src/main/java/com/dotmarketing/startup/StartupTasksExecutor.java

```diff
@@ -1,343 +1,200 @@
-				return o1.getName().compareTo(o2.getName());
-			}
-		};
-		try {
-			runOnce = TaskLocatorUtil.getStartupRunOnceTaskClasses();
-			runAlways = TaskLocatorUtil.getStartupRunAlwaysTaskClasses();
-		} catch (Exception e) {
-			throw new DotRuntimeException(e.getMessage(), e);
-		}
-		Collections.sort(runOnce, comparator);
-		Collections.sort(runAlways, comparator);
-		Logger.debug(this.getClass(), "Locking db_version table");
-		setupSQL();
-		//Integer currentVersion = null;
-		PreparedStatement update = null;
-		Statement s = null;
-		ResultSet rs = null;
-		Connection conn = null;
-		try {
-			conn = DbConnectionFactory.getDataSource().getConnection();
+	public static StartupTasksExecutor getInstance() {
+		if (executor == null)
+			executor = new StartupTasksExecutor();
+		return executor;
+	}
 
-			conn.setAutoCommit(false);
-			s = conn.createStatement();
-			update = conn
-					.prepareStatement("INSERT INTO db_version (db_version,date_update) VALUES (?,?)");
-			s.execute(lock);
-			rs = s
-					.executeQuery(select);
-			rs.next();
-			Config.DB_VERSION = rs.getInt("db_version");
-		} catch (SQLException e) {
-			// Maybe the table doesn't exist?
-			Logger.debug(this.getClass(), "Trying to create db_version table");
-			try {
-				conn.rollback();
-				if(DbConnectionFactory.isMySql()){
-					s.execute("SET " + DbConnectionFactory.getMySQLStorageEngine() + "=INNODB");
-				}
-				s.execute(create);
-				if(update==null) {
-				    // looks like H2 do an early table name check
-				    update = conn.prepareStatement("INSERT INTO db_version (db_version,date_update) VALUES (?,?)");
+	/**
+	 * Check which database we're using, and select the apropiate SQL. In a
+	 * different method to avoid further clutter
+	 * @throws DotDataException 
+	 */
+	private void setupDBVersionTable() throws DotDataException {
+		try{
+			LocalTransaction.wrap(()->{
+				Connection conn  = DbConnectionFactory.getConnection();
+				ResultSet rs;
+				try {
+					rs = conn.prepareStatement(select).executeQuery();
+					rs.next();
+					Config.DB_VERSION = rs.getInt("db_version");
+				} catch (SQLException e) {
+					throw new DotDataException(e);
 				}
-				update.setInt(1, 0);
-				Date date = new Date(Calendar.getInstance().getTimeInMillis());
-				update.setDate(2, date);
-				update.execute();
-				conn.commit();
-
-				Logger
-						.debug(this.getClass(),
-								"Table db_version created.  Trying to lock db_table again.");
-				s.execute(lock);
-				rs = s
-						.executeQuery(select);
-
-				rs.next();
-				Config.DB_VERSION = rs.getInt("db_version");
-
-			} catch (SQLException e2) {
-				Logger.fatal(this.getClass(),
-						"Locking of db_version table failed: "
-								+ e2.getMessage());
-				throw new DotRuntimeException(
-						"Locking of db_version table failed: "
-								+ e2.getMessage(), e2);
-			}
 
+			});
+		} catch (DotDataException e) {
+			LocalTransaction.wrap(()->{
+				Config.DB_VERSION = 0;
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2696)